components/brave_ads/browser/ads_service_impl.cc

```diff
@@ -0,0 +1,914 @@
+  // TODO - implement this
+  return true;
+}
+
+void AdsServiceImpl::OnShow(Profile* profile,
+                            const std::string& notification_id) {
+  if (!connected() ||
+      notification_ids_.find(notification_id) == notification_ids_.end())
+    return;
+
+  bat_ads_->GenerateAdReportingNotificationShownEvent(
+      notification_ids_[notification_id]->ToJson(), base::BindOnce(&Noop));
+}
+
+void AdsServiceImpl::OnClose(Profile* profile,
+                             const GURL& origin,
+                             const std::string& notification_id,
+                             bool by_user,
+                             base::OnceClosure completed_closure) {
+  if (notification_ids_.find(notification_id) != notification_ids_.end()) {
+    auto notification_info = base::WrapUnique(
+        notification_ids_[notification_id].release());
+    notification_ids_.erase(notification_id);
+
+    if (connected()) {
+      auto result_type = by_user
+          ? ads::NotificationResultInfoResultType::DISMISSED
+          : ads::NotificationResultInfoResultType::TIMEOUT;
+      bat_ads_->GenerateAdReportingNotificationResultEvent(
+          notification_info->ToJson(),
+          ToMojomNotificationResultInfoResultType(result_type),
+          base::BindOnce(&Noop));
+    }
+  }
+
+  std::move(completed_closure).Run();
+}
+
+void AdsServiceImpl::OpenSettings(Profile* profile,
+                                  const GURL& origin) {
+  DCHECK(origin.has_query());
+  auto notification_id = origin.query();
+
+  if (notification_ids_.find(notification_id) == notification_ids_.end())
+    return;
+
+  auto notification_info = base::WrapUnique(
+      notification_ids_[notification_id].release());
+  notification_ids_.erase(notification_id);
+
+  if (connected()) {
+    bat_ads_->GenerateAdReportingNotificationResultEvent(
+        notification_info->ToJson(),
+        ToMojomNotificationResultInfoResultType(
+            ads::NotificationResultInfoResultType::CLICKED),
+        base::BindOnce(&Noop));
+  }
+
+  GURL url(notification_info->url);
+
+  Browser* browser = chrome::FindLastActiveWithProfile(profile);
+  NavigateParams nav_params(browser, url, ui::PAGE_TRANSITION_LINK);
+  nav_params.disposition = WindowOpenDisposition::NEW_FOREGROUND_TAB;
+  // TODO(bridiver) - what to put here?
+  // nav_params.referrer = GURL("https://brave.com");
+  nav_params.window_action = NavigateParams::SHOW_WINDOW;
+  Navigate(&nav_params);
+}
+
+void AdsServiceImpl::GetClientInfo(ads::ClientInfo* client_info) const {
+  // TODO(bridiver) - these eventually get used in a catalog request
+  // and seem like potential privacy issues
+  // client_info->application_version = "";
+  // client_info->platform_version = "";
+  // client_info.application_version = chrome::kChromeVersion;
+#if defined(OS_MACOSX)
+  client_info->platform = ads::ClientInfoPlatformType::MACOS;
+#elif defined(OS_WIN)
+  client_info->platform = ads::ClientInfoPlatformType::WIN10;
```

Comment:
Is it not possible to deter win7 and win8 as if it we need to discuss with Mandar to change spec

Reply:
I wouldn't let this pass privacy review with that info so I didn't want to add it and then leave a comment on my own PR that it's not ok :)